api/src/main/java/org/apache/iceberg/Table.java

```diff
@@ -206,6 +206,12 @@ default AppendFiles newFastAppend() {
    */
   Rollback rollback();
```

Comment:
Can we combine this with the Rollback API? We could still support the `rollback` method here, but combine `Rollback` and `CherryPick` into something like `ManageSnapshots`. Then we could reuse logic for enforcing checks of the current snapshot. What do you think?

Reply:
@rdblue Does this mean, we would have - 

```
table.manageSnapshot().cherrypick(snapshotId)
table.manageSnapshot().rollback(snapshotId)
``` 

instead of current implementation -

```
table.cherrypick().cherrypick(snapshotId)
table.rollback().toSnapshotId(snapshotId)
```

Reply:
Yes, something like that. We might want to deprecate the `rollback` API and just have one for managing snapshots that can do different tasks:

```java
table.manageSnapshots()
    .cherrypick(id1)
    .cherrypick(id2)
    .commit()
```
```java
table.manageSnapshots()
    .fastForwardCurrentTo(id)
    .commit()
```
```java
table.manageSnapshots()
    .resetCurrentTo(id)
    .commit()
```

I originally thought we may want to make it API compatible with `rollback` but `manageSnapshots().toSnapshotId(id)` doesn't make much sense.